### PR TITLE
busybox: drop NFS support for linux < 2.6.23 (enable NFSv4)

### PR DESCRIPTION
**Pull request by MilhouseVH**
This enables NFSv4 support.

Current nightly:
```
LibreELEC (community): devel-20180921003158-#0920-gb6346ba (Generic.x86_64)
NUC:~ # mkdir /tmp/nfs
NUC:~ # mount -t nfs -o vers=4 192.168.0.9:/nfs /tmp/nfs
mount: NFSv4 not supported
mount: mounting 192.168.0.9:/nfs on /tmp/nfs failed
NUC:~ # exit
```

Build with this PR:
```
LibreELEC (community): devel-20180921164124-#0921b-g59c1e54 (Generic.x86_64)
NUC:~ # mkdir /tmp/nfs
NUC:~ # mount -t nfs -o vers=4 192.168.0.9:/nfs /tmp/nfs
NUC:~ # ls -la /tmp/nfs
total 24
drwxrwxr-x    7 root     root          4096 Mar 22  2018 .
drwxrwxrwt   10 root     root           220 Sep 21 18:04 ..
drwxrwxr-x    3 root     root          4096 Apr  4 01:59 Storage
drwxr-xr-x    4 root     root          4096 Jul 26 05:37 System
drwxr-xr-x    4 root     root          4096 Aug  9  2016 System.bak
drwxrwxr-x    4 root     root          4096 Aug 10  2016 System.old
drwxr-xr-x    4 root     root          4096 Mar 22  2018 System.oldpi3
NUC:~ # mount | grep /tmp/nfs
192.168.0.9:/nfs on /tmp/nfs type nfs4 (rw,relatime,vers=4.0,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=192.168.0.20,local_lock=none,addr=192.168.0.9)
```

If `-o vers=4` is not specified, LibreELEC will connect with NFSv3 (when testing against Ubuntu 17.10 with an NFSv4 server, and using `watch -n1 nfsstat -sl#` on the server - where # is 3 or 4 - to see which counters increased, either 3 or 4, while copying a file from the server to the client).

### CODE REVIEW

**Comment by InuSasha** [comment]
Will this drop the possibility to boot with storage from NFS, too?

File: packages/sysutils/busybox/config/busybox-init.conf [hunk 1]
```diff
@@ -33,7 +33,7 @@ CONFIG_FEATURE_PREFER_APPLETS=y
 CONFIG_BUSYBOX_EXEC_PATH="/proc/self/exe"
 # CONFIG_SELINUX is not set
 # CONFIG_FEATURE_CLEAN_UP is not set
-CONFIG_FEATURE_SYSLOG=y
+# CONFIG_FEATURE_SYSLOG is not set
```
Comment (chewitt) [inline packages/sysutils/busybox/config/busybox-init.conf]: intentional change?

Comment (MilhouseVH) [reply packages/sysutils/busybox/config/busybox-init.conf]: Yes, busybox `make oldconfig`  disabled it - syslog is only being enabled by [FEATURE_MOUNT_NFS](https://github.com/mirror/busybox/blob/1303962957fb900ed97c5958403990e885b06e29/util-linux/mount.c#L68-L79):
```
//config:config FEATURE_MOUNT_NFS
//config:	bool "Support mounting NFS file systems on Linux < 2.6.23"
//config:	default n
//config:	depends on MOUNT
//config:	select FEATURE_SYSLOG
//config:	help
//config:	Enable mounting of NFS file systems on Linux kernels prior
//config:	to version 2.6.23. Note that in this case mounting of NFS
//config:	over IPv6 will not be possible.
//config:
//config:	Note that this option links in RPC support from libc,
//config:	which is rather large (~10 kbytes on uclibc).
```

**Comment by MilhouseVH** [comment]
> Will this drop the possibility to boot with storage from NFS, too?

No, I don't believe so - it's just dropping support for ancient kernels that we don't support. I'll test network booting over NFS later today.

**Comment by MilhouseVH** [comment]
Booting from NFS still works (tested RPi3+ booting #0923, with both boot= and disk= on NFS). No reported issues with this change, so looks good to me.

**Pull request merged**